### PR TITLE
feat: expose rowHeight from onHeightChange

### LINKED ISSUE #271: Expose row count information
Hi,

we're trying to migrate to v8 but we realized that the change in `onHeightChange` function arguments is causing us to miss on a key feature.

**Background**

In our UI components library we have a component called `MultilineTextInput`, which uses the resizable textarea component. One of the features of this component is that when there is value in more than one line, we show an expand/collapse button.

You can see a live example here: https://uikit.commercetools.com/?path=/story/components-inputs--multilinetextinput

To do that, we rely on the `valueRowCount` exposed by the component.

https://github.com/commercetools/ui-kit/blob/86a85e642d0d47089c3ddd7094974c58badbed56/src/components/inputs/multiline-text-input/multiline-text-input.js#L30-L40

In v8 though, the `onHeightChange` only exposes the `height` value.

Ideally, what we need is the internally calculated `rowHeight`

https://github.com/Andarist/react-textarea-autosize/blob/757355da9d9810b7855621b0093cf26e323d48cd/src/calculateNodeHeight.ts#L50

And to calculate the row count:

```js
const valueRowCount = Math.floor(height / rowHeight);
```

**Proposal**

I'm not sure what the best solution here would be. Maybe we can introduce a new callback prop that specifically exposes this value.

What do you think? Do you see any other solution?

I'm happy to work on a PR if we decide how to proceed.

Thanks!

### PR DESCRIPTION
**Pull request by emmenko**
Fixes #271

### CODE REVIEW

File: src/calculateNodeHeight.ts [hunk 1]
```diff
@@ -61,5 +66,5 @@ export default function calculateNodeHeight(
   }
   height = Math.min(maxHeight, height);
 
-  return height;
+  return { height, rowHeight };
```
Comment (Andarist) [inline src/calculateNodeHeight.ts:69]: could we make this a tuple? the final code would be slightly smaller 😉 TS 4.0 will come for labels for tuples so we'll be able to improve readability for this then

File: .changeset/fair-bobcats-matter.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (Andarist) [inline .changeset/fair-bobcats-matter.md:1]: Could you also update the README?

File: src/calculateNodeHeight.ts (hunk 1 above)
Comment (emmenko) [reply src/calculateNodeHeight.ts:69]: Sure, I wasn't sure if you wanted to expose more values.

File: .changeset/fair-bobcats-matter.md (hunk 2 above)
Comment (emmenko) [reply .changeset/fair-bobcats-matter.md:1]: The description of the `onHeightChange` in the table?

Comment (Andarist) [reply .changeset/fair-bobcats-matter.md:1]: Ye

File: .changeset/fair-bobcats-matter.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+'react-textarea-autosize': minor
+---
+
+Expose `{ rowHeight: number }` as second argument of the callback function `onHeightChange`. This is useful to construct custom behaviors according to the height values.
```
Comment (Andarist) [inline .changeset/fair-bobcats-matter.md:5]:
```suggestion
Added `{ rowHeight: number }` as a second parameter to the `onHeightChange` callback. This is useful to construct custom behaviors according to the height values.
```

**Pull request merged**